### PR TITLE
GT-363 Improve ArangoDB version detection

### PR DESCRIPTION
**Pull request by nikita-vanyasin**
`log.force-direct` option forces arangod to write output using main thread - ensuring the whole log output will be flushed before return

Also some refactoring of DatabaseVersion to improve readability.

The retry logic probably can be deleted, but let's wait if this warning will disappear completely.

### CODE REVIEW

File: service/version_check.go [hunk 1]
```diff
@@ -31,34 +31,36 @@ import (
...
+	for i := 0; i < retries; i++ {
+		var v driver.Version
+		var enterprise bool
+		v, enterprise, err = s.databaseVersion(ctx)
+		if err == nil {
+			return v, enterprise, nil
 		}
 
-		return d, enterprise, nil
+		s.log.Warn().Err(err).Msgf("Error while getting version. Attempt %d of %d", i+1, retries)
```
Comment (informalict) [inline service/version_check.go:52]: if ctx.Err() != nil {
// maybe SIGTERM was sent to the starter so this context probably is closed then, and `DatabaseVersion` should return ASAP
}

**Review by informalict — COMMENTED** [review]
lgtm with one comment

File: service/version_check.go (hunk 1 above)
Comment (nikita-vanyasin) [reply service/version_check.go:52]: The reason is known for long time - arangod by default uses background thread for writing the output. Main program can terminate before the background thread will flush the buffer. Added flag changes that.

**Review by informalict — APPROVED** [review]
lgtm

**Pull request merged**